apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -24,6 +23,232 @@ namespace Google.Cloud.Bigtable.V2
+        /// </param>
+        /// <returns>
+        /// The response from trying to mutate the row.
+        /// </returns>
+        public virtual Task<CheckAndMutateRowResponse> CheckAndMutateRowAsync(
+            TableName tableName,
+            BigtableByteString rowKey,
+            RowFilter predicateFilter,
+            IEnumerable<Mutation> trueMutations,
+            IEnumerable<Mutation> falseMutations,
+            CallSettings callSettings = null) =>
+            CheckAndMutateRowAsync(
+                CreateCheckAndMutateRowRequest(tableName, rowKey, predicateFilter, trueMutations, falseMutations),
+                callSettings);
+
+        /// <summary>
+        /// Asynchronously mutates a row atomically based on the output of a predicate Reader filter.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that string is implicitly convertible to <see cref="BigtableByteString"/>, so <paramref name="rowKey"/> can
+        /// be specified using a string as well and its UTF-8 representations will be used.
+        /// </para>
+        /// <para>
+        /// This method simply delegates to <see cref="CheckAndMutateRow(CheckAndMutateRowRequest, CallSettings)"/>.
+        /// </para>
+        /// </remarks>
+        /// <param name="tableName">
+        /// The unique name of the table to which the conditional mutation should be
+        /// applied. Must not be null.
+        /// </param>
+        /// <param name="rowKey">
+        /// The key of the row to which the conditional mutation should be applied.
+        /// Must not be empty.
+        /// </param>
+        /// <param name="predicateFilter">
+        /// The filter to be applied to the contents of the specified row. Depending
+        /// on whether or not any results are yielded, <paramref name="trueMutations"/> may or may not
+        /// be executed. If null, checks that the row contains any values at all.
+        /// </param>
+        /// <param name="trueMutations">
+        /// Changes to be atomically applied to the specified row if <paramref name="predicateFilter"/>
+        /// yields at least one cell when applied to <paramref name="rowKey"/>. Entries are applied in
+        /// order, meaning that earlier mutations can be masked by later ones.
+        /// Must contain at least one entry and at most 100000.  Must not be null, or contain null elements.
+        /// </param>
+        /// <param name="cancellationToken">A cancellation token to monitor for the asynchronous operation.</param>
+        /// <returns>
+        /// The response from trying to mutate the row.
+        /// </returns>
+        public virtual Task<CheckAndMutateRowResponse> CheckAndMutateRowAsync(
+            TableName tableName,
+            BigtableByteString rowKey,
+            RowFilter predicateFilter,
+            IEnumerable<Mutation> trueMutations,
+            CancellationToken cancellationToken = default) =>
```

Comment:
It feels slightly odd that whether or not you can specify a cancellation token or a call settings depends on whether you're providing `falseMutations`.

I can see the following potential overloads (and there could be more that I haven't thought about):

```csharo
// No settings/cancellation token due to params
CAMRA(tableName, rowKey, predicateFilter, params trueMutations)
CAMRA(tableName, rowKey, predicateFilter, trueMutations, token)
CAMRA(tableName, rowKey, predicateFilter, trueMutations, callSettings = null)
CAMRA(tableName, rowKey, predicateFilter, trueMutations, falseMutations, token)
CAMRA(tableName, rowKey, predicateFilter, trueMutations, falseMutations, callSettings = null)
```

That creates an ambiguity if anyone calls `CAMRA(tableName, rowKey, predicateFilter, trueMutations, null)` as the null could be call settings or `falseMutations`.

Do we know how common "true mutations only" is? Does it deserve its own method name to avoid overloading problems?

Reply:
Not sure about usage. I'll ask internally. Good call about oddness. I guess I had these as mirrors to the synchronous params overloads, but kept the cancellation token because it is used for the async pattern, which meant they couldn't be params, so we lost the benefit there. What about just these two overloads:
```
CAMRA(tableName, rowKey, predicateFilter, params trueMutations)
CAMRA(tableName, rowKey, predicateFilter, trueMutations, falseMutations, callSettings = null)
```

Reply:
That's probably a reasonable pair, yes. Let's at least start with that...

Reply:
Oops, I made the comment and then forgot to actually make this change. Hold on I'll change this now.